Feature/api-ut

asv-tungpt opened the pull request.

asv-tungpt committed:
Create api

asv-tungpt committed:
Submit api

asv-tungpt committed:
Update file

asv-tungpt committed:
Test api successful

asv-tungpt committed:
Remove unnecessary post method

asv-tungpt committed:
Repair some issues

asv-tungpt committed:
unit test

phucnpm commented on djangoguestbook/guestbook/test_model/test.py:
```diff
@@ -0,0 +1,235 @@
+		assert greetings is not None and len(greetings) == 20
+
+	def test_get_lastest_with_in_of_size(self):
+		greetings = Greeting.get_lastest(self.guestbook_name, 5, False)
```
hãy dùng mock để fake giá trị trả về của memcache.get( ) và phân làm 2 trường hợp test
- có cache
- không có cache

phucnpm replied:
nếu có cache thì mình phải test xem function _query_update_memcache có đc gọi hay không ? mock làm dc việc này

phucnpm commented on djangoguestbook/guestbook/test_model/test.py:
```diff
@@ -0,0 +1,235 @@
+		assert greetings is not None and len(greetings) == 5
+
+	def test__query_update_memcache(self):
+		greetings = Greeting._query_update_memcache(self.guestbook_name, 10)
```
với đoạn code này thì không test được là cache có update hay không

phucnpm commented on djangoguestbook/guestbook/test_model/test.py:
```diff
@@ -0,0 +1,235 @@
+			'author': "test_author",
+			'content': "testing"
+		}
+		greeting = Greeting.put_from_dict(dict)
```
test get wrong id, thì không cần thiết phải tạo ra 1 greeting đúng rồi mới tạo wrong_id

phucnpm commented on djangoguestbook/guestbook/test_model/test.py:
```diff
@@ -0,0 +1,235 @@
+		got_greeting = Greeting.get_greeting_by_id(self.guestbook_name, greeting.key.id())
+		assert got_greeting is None
+
+	def test_delete_greeting_wiht_right_id(self):
```
2 hàm test_delete_greeting_wiht_right_id(self): ?

phucnpm commented on djangoguestbook/guestbook/test_model/test.py:
```diff
@@ -0,0 +1,235 @@
+		greeting.date = datetime.strptime('2015-02-11 13:00 +0000', "%Y-%m-%d %H:%M +0000")
+		greeting.updated_by = "updated_author"
+		greeting.updated_date = datetime.strptime('2015-02-11 13:00 +0000', "%Y-%m-%d %H:%M +0000")
+		greeting.put()
```
thao tác put( ) là không cần thiết

asv-tungpt replied:
ko đặt put() thì nó ko có id để get().

phucnpm replied:
> ko đặt put() thì nó ko có id để get().

test greeting.to_dict() thì không cần thiết phải get()

asv-tungpt replied:
em vẫn chưa hiểu ý anh, em test greeting.to_dict() đâu có get() gì đâu?

phucnpm commented on djangoguestbook/guestbook/test_model/test.py:
```diff
@@ -0,0 +1,235 @@
+			'updated_by': "updated_author",
+			'updated_date': "2015-02-11 07:11 +0000"
+		}
+		assert test_dict != greeting.to_dict()
```
thao tác test bằng cách compare với 1 dictionary khác là không cần thiết

phucnpm commented on djangoguestbook/guestbook/test_model/test.py:
```diff
@@ -0,0 +1,235 @@
+			'updated_by': "update_author",  # wrong updated author
+			'updated_date': "2015-02-11 07:11 +0000"
+		}
+		assert test_dict != greeting.to_dict()
```
thao tác test bằng cách compare với 1 dictionary khác là không cần thiết

phucnpm commented on djangoguestbook/guestbook/test_model/test.py:
```diff
@@ -0,0 +1,235 @@
+
+	def test_get_greeting_with_no_cursor_in_of_size(self):
+		greetings, next_cursor, next = Greeting.get_greeting(self.guestbook_name, 10, None)
+		assert greetings is not None and next_cursor is not None and next is True
```
logic test này không có ý nghĩa, tại sao `cursor==None` thì `next_cursor is not None and next is True` ?

phucnpm commented on djangoguestbook/guestbook/test_model/test.py:
```diff
@@ -0,0 +1,235 @@
+		assert test_dict != greeting.to_dict()
+
+	def test_get_greeting_with_no_cursor_in_of_size(self):
+		greetings, next_cursor, next = Greeting.get_greeting(self.guestbook_name, 10, None)
```
không nên thay đổi tên biến khác với define của function
đổi `next` lại thành `more`

phucnpm replied:
logic test hãy cập nhật lại 
- Check số lượng greeting trong DB `total_count`
  - Nếu số `num_greetings` < `total_count` thì `more` phải là true và ngc lại

asv-tungpt replied:
Anh phúc, làm sao để check được số lượng greeting trong DB?

phucnpm replied:
cách đơn giản nhất là em query all greeting (no limit) rồi get length ( len() )

asv-tungpt committed:
Submit unit test

Em chưa import được: form unittest.mock import patch
Nên vẫn chưa test được trường hợp cache và non

asv-tungpt committed:
Using mock

asv-tungpt committed:
check update memcache
